scripts/convert-to-safari-rules.ts

```diff
@@ -1,14 +1,276 @@
-import { loadAllLists } from '../assets/load';
-import { convertCosmetics, convertFilter } from '../src/convertion/safari-rules';
+import { fetchLists } from './fetch-lists';
 import { parseList } from '../src/parsing/list';
+import { convertCosmetics, convertFilter } from '../src/convertion/safari-rules';
 
-function main() {
-  const { networkFilters, cosmeticFilters } = parseList(loadAllLists());
+const allowedTriggerKeys = new Set([
+  'url-filter', 'url-filter-is-case-sensitive', 'if-domain', 'unless-domain', 'resource-type', 'load-type', 'if-top-url', 'unless-top-url',
+]);
+
+const allowedActionKeys = new Set([
+  'type', 'selector',
+]);
+
+const allowedResourceTypes = new Set([
+  'document', 'image', 'style-sheet', 'script', 'font', 'raw', 'svg-document', 'media', 'popup',
+]);
+
+const allowedLoadTypes = new Set(['first-party', 'third-party']);
+
+const allowedActions = new Set(['block', 'block-cookies', 'css-display-none', 'ignore-previous-rules', 'make-https']);
+
+function checkObjectKeysAllowed(keys, allowedKeys) {
+  for (let i = 0; i < keys.length; i += 1) {
+    isSame(allowedKeys.has(keys[i]), true, 'key not allowed');
```

Comment:
Not sure about using `isSame` here. Feels like it defeats the purpose, when it might be enough to do:
```js
if (!allowedKeys.has(keys[i])) {
 throw msg;
}
```